HW4_Vaganova/protein_analysis_tool.py

```diff
@@ -95,7 +119,21 @@ def find_site(seq: str, site: str) -> str:
     :param site: specify site of interest as short seq in 1-latter code (str)
     :return: positions of residues for each certain site in seq (str)
     """
-    pass
+    if seq not in RESIDUES_NAMES.values():
+        raise ValueError(f'{site} site is not a protein sequence!')
+    if site in seq:
+        site_full_coordinates = []
+        site_count = seq.count(site)
+        site_start_coordinates = [coordinate for coordinate in range(len(seq)) if seq.startswith(site, coordinate)]
+        site_end_coordinates = [(coordinate + len(site)) for coordinate in site_start_coordinates]
+        for counter in range(len(site_start_coordinates)):
+            site_full_coordinates.append([site_start_coordinates[counter], site_end_coordinates[counter]])
+        if site_count == 1:
+            return f'{site} found in sequence {site_count} time; site coordinates are {site_full_coordinates}'
+        else:
+            return f'{site} found in sequence {site_count} times; site coordinates are {site_full_coordinates}'
```

Comment:
Difference between two return statements is unclear. Please, make it more obvious.